File: functions/voice/tts.js
// Dev 5 — implement this handler
// Converts agent text response to MP3 audio via Amazon Polly (neural engine).
// The browser decodes the base64 MP3 and plays it to the user.
//
// Switch to Python 3.12 runtime when implementing.
//
// Expected input:
//   { text: string }
//
// Returns:
//   { statusCode: 200, body: { result: "Audio synthesized", data: { audioBase64, contentType: "audio/mpeg" } } }
//   { statusCode: 400, body: { message: "No text provided" } }
//
// Env vars: AWS_REGION (set by Lambda runtime), POLLY_VOICE_ID (optional, default Joanna)
// IAM: polly:SynthesizeSpeech granted via VoiceLambdaRole in FunctionsStack

exports.handler = async (event) => {
  throw new Error('Not implemented — Dev 5 owns this function');
};

File: functions/voice/tts.py
import base64
import json
import os

import boto3

polly = boto3.client("polly", region_name=os.environ.get("AWS_REGION", "us-east-1"))

VOICE_ID = os.environ.get("POLLY_VOICE_ID", "Joanna")
ENGINE = os.environ.get("POLLY_ENGINE", "neural")

HEADERS = {
    "Content-Type": "application/json",
    "Access-Control-Allow-Origin": "*",
    "Access-Control-Allow-Headers": "Content-Type,Authorization",
    "Access-Control-Allow-Methods": "POST,OPTIONS",
}


def response(status_code: int, payload: dict) -> dict:
    return {
        "statusCode": status_code,
        "headers": HEADERS,
        "body": json.dumps(payload),
    }


def handler(event, context):
    if event.get("httpMethod") == "OPTIONS":
        return response(200, {})

    try:
        body = json.loads(event.get("body") or "{}")
        text = body.get("text", "")

        if not text.strip():
            return response(400, {"message": "No text provided"})

        synth = polly.synthesize_speech(
            Text=text,
            OutputFormat="mp3",
            VoiceId=body.get("voiceId", VOICE_ID),
            Engine=body.get("engine", ENGINE),
        )
        audio_bytes = synth["AudioStream"].read()

        return response(
            200,
            {
                "result": "Audio synthesized",
                "data": {
                    "audioBase64": base64.b64encode(audio_bytes).decode("utf-8"),
                    "contentType": "audio/mpeg",
                    "voiceId": body.get("voiceId", VOICE_ID),
                    "engine": body.get("engine", ENGINE),
                },
            },
        )
    except Exception as exc:
        return response(500, {"message": str(exc)})
